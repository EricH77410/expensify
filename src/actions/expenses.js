
import database from '../firebase/firebase';

// ADD_EXPENSE
export const addExpense = (expense) => {
    return {
        type: 'ADD_EXPENSE',
        expense
    }
}

export const startAddExpense = (expenseData = {})=>{
   return (dispatch)=>{
    const {
        description='',
        note='',
        amount=0,
        createdAt=0
    } = expenseData
    const expense = {description, note, amount, createdAt}
    
    return database.ref('expenses').push(expense)
        .then((snap)=>{dispatch(addExpense({
            id: snap.key,
            ...expense
        }))})
   }
}
// REMOVE_EXPENSE
export const removeExpense = ({id}) => {
    return {
        type:'REMOVE_EXPENSE',
        id
    }
}
// ASYNC
export const startRemoveExpense = ({id})=>{
    return (dispatch) => {
        return database.ref(`expenses/${id}`).remove()
            .then(()=>{
                dispatch(removeExpense({id}))
            })
    }
}

// EDIT_EXPENSE
export const editExpense = (id,update) => {
    return {
        type: 'EDIT_EXPENSE',
        id,
        update
    }
}

// SET EXPENSES
export const setExpenses = (expenses) => {
    return {
        type: 'SET_EXPENSES',
        expenses
    }
}

export const startSetExpenses = () => {
    return (dispatch) => {
        const expenses = [];

        return database.ref('expenses').once('value').then((snap)=>{
            snap.forEach((childSnap)=>{
                expenses.push({
                    id:childSnap.key,
                    ...childSnap.val()
                })
            })
        }).then(()=>{
            dispatch(setExpenses(expenses));
        })
    }
}


