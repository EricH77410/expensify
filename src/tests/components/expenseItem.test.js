import React from 'react'
import { shallow } from 'enzyme'
import ExpenseItem from '../../components/ExpenseItem';

import expenses from '../fixtures/expenses';

test('should render an expense item correctly', () =>{
   const wrapper = shallow(<ExpenseItem {...expenses[1]}/>);
   expect(wrapper).toMatchSnapshot();
})