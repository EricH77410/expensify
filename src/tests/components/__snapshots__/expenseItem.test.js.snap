// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an expense item correctly 1`] = `
<div>
  <Link
    replace={false}
    to="edit/2"
  >
    <h3>
      Cake
    </h3>
  </Link>
  <p>
    $103.90
    -
    December 28th, 1969
  </p>
  <hr />
</div>
`;
